<?xml version="1.0" ?>
<robot name="car_model">
 
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>

  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>

  <link name="base_footprint"/>

  <joint name="base_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link" />
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>

  <link name="base_link">
    <visual>
      <origin xyz="1.1 0 0.85" rpy="0 0 0"/>
      <geometry>
        <box size="3.332 1.28 1.5"/>
      </geometry>
      <material name="white"/>
    </visual>

    <collision>
      <origin xyz="1.1 0 0.85" rpy="0 0 0"/>
      <geometry>
        <box size="3.332 1.28 1.5"/>
      </geometry>
    </collision>

    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <mass value="8.1e+02"/>
      <inertia ixx="302" ixy="0" ixz="0"
               iyy="893" iyz="0"
               izz="891" />
    </inertial>
  </link>

  <joint name="base_to_steering_point" type="continuous">
    <parent link="base_link"/>
    <child link="steering_point"/>
    <axis xyz="0 0 1"/>
    <origin xyz="2.2 0 0"/>
  </joint>

  <link name="steering_point"/>

  <joint name="steering_point_to_front_wheel" type="continuous">
    <parent link="steering_point"/>
    <child link="front_wheel_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.57075 0 0"/>
  </joint>

  <link name="front_wheel_link">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 1.57075"/>
      <geometry>
        <cylinder length="0.114" radius="0.256"/>
      </geometry>
      <material name="blue"/>
    </visual>

    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.114" radius="0.256"/>
      </geometry>
      <material name="blue"/>
    </collision>

    <inertial>
      <origin xyz="0 0 0" />
      <mass value="15" />
      <inertia ixx="0.262005" ixy="0" ixz="0"
               iyy="0.262005" iyz="0"
               izz="0.49152" />
      </inertial>
  </link>

  <joint name="back_wheel_left_joint" type="continuous">
    <parent link="base_link"/>
    <child link="back_wheel_left_link"/>
    <origin xyz="0.0 0.64 0" rpy="1.57075 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>

  <link name="back_wheel_left_link">
    <visual>
      <origin xyz="0 0 0.057" rpy="0 0 1.57075"/>
      <geometry>
        <cylinder length="0.114" radius="0.256"/>
      </geometry>
      <material name="blue"/>
    </visual>

    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.114" radius="0.256"/>
      </geometry>
    </collision>

    <inertial>
      <origin xyz="0 0 0" />
      <mass value="15" />
      <inertia ixx="0.262005" ixy="0" ixz="0"
               iyy="0.262005" iyz="0"
               izz="0.49152" />
      </inertial>
  </link>

  <joint name="back_wheel_right_joint" type="continuous">
    <parent link="base_link"/>
    <child link="back_wheel_right_link"/>
    <origin xyz="0.0 -0.64 0" rpy="-1.57075 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>

  <link name="back_wheel_right_link">
    <visual>
      <origin xyz="0 0 -0.057" rpy="0 0 1.57075"/>
      <geometry>
        <cylinder length="0.114" radius="0.256"/>
      </geometry>
      <material name="blue"/>
    </visual>

    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.114" radius="0.256"/>
      </geometry>
    </collision>

    <inertial>
      <origin xyz="0 0 0" />
      <mass value="15" />
      <inertia ixx="0.262005" ixy="0" ixz="0"
               iyy="0.262005" iyz="0"
               izz="0.49152" />
      </inertial>
  </link>

</robot>
